data_visualization_in_open_source_software/data_visualization_in_open_source_software.md

```diff
@@ -0,0 +1,286 @@
+
+Analysts frequently find themselves in situations where they want to depict more than two dimensions at a time. While it is possible to create 3D plots and animated plots, there are often simpler ways to express additional dimensions --- and simpler is nearly always better when it comes to visualizations!
+
+You can use **color** to communicate either continuous (e.g. darker = higher value) or categorical values (e.g. red = "no" and blue = "yes"). **Shape** (e.g. circle, triangle, cross) or **line type** (e.g. solid or dashed) can also indicate category membership and have the advantage of not relying on end users' ability to distinguish color. **Size** is often used to indicate count or population values.
+
+![A scatter plot of age vs glucose with lines of best fit, with patients in Class 1 shown as yellow dots with a solid best fit line, and Class 2 as blue triangles with a blue dashed line.](media/intro_2.png)
+
+Another valuable tool is **faceting**. This is when similar plots (that share one or both axes) are lined up next to each other to facilitate comparison. For example, when you want to show the relationship between two variables, X and Y, for several different subsets of the data, you can generate a scatter plot of X and Y for each subset of the data as its own facet. If you have more than two or three groups, faceting generally produces clearer plots than using color or shape to distinguish groups.
+
+![Two scatter plots side by side showing age vs glucose levels for patients in Class 1 (left) and Class 2 (right). The y-axis is aligned to facilitate comparison across groups.](media/intro_3.png)
+
+Most of these plot elements can be applied across a wide range of kinds of visualizations. Knowing how to identify these key plot elements will help you select effective visualizations for your own data.
+
+Before we dive into how plotting works in practice, though, we'll explore some popular open source tools for data visualization.
+
+## Plotting libraries in R and python
+
+There are many options available for creating data visualizations in R and python. For this module, we'll focus on just one plotting library in each: `ggplot2` in R, and `seaborn` in python.
+
+### `ggplot2` in R
+
+R comes with plenty of plotting functions ready to go in base R, and for quick-and-dirty visualizations, that is usually the fastest option. The popular library [`ggplot2`](https://ggplot2.tidyverse.org/) provides much more flexibility and control over your visualizations, however, and if you're creating visualizations to share (e.g. in a manuscript or presentation), you'll probably find you need to switch to `ggplot2` at some point anyway to get more control over the appearance of your visualizations.
+
+To get started with `ggplot2` in R on your own machine, you'll need to install it, and then load the library in R:
+
+```r
+# You only need to install it once
+install.packages("ggplot2")
+
+# You'll need to load the library anew for each R session
```

Comment:
```suggestion
# You'll need to load the library again for each R session
```
not particularly common word, not sure if it may introduce language barriers for some